homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -422,3 +429,129 @@ def values_to_watch(self, resolved_data: dict[str, Any]) -> Iterable[ZwaveValue]
     def current_tilt_value(resolved_data: dict[str, Any]) -> ZwaveValue | None:
         """Get current tilt ZwaveValue from resolved data."""
         return resolved_data["tilt_value"]
+
+
+@dataclass
+class FanSpeedDataTemplate:
+    """Mixin to define get_speed_config."""
+
+    def get_speed_config(self, resolved_data: dict[str, Any]) -> list[int] | None:
+        """
+        Get the fan speed configuration for this device.
+
+        Values should indicate the highest allowed device setting for each
+        actual speed, and should be sorted in ascending order.
+
+        Empty lists are not permissible.
+        """
+        # pylint: disable=no-self-use
+        raise NotImplementedError
+
+
+@dataclass
+class FixedFanSpeedValueMix:
+    """Mixin data class for defining supported fan speeds."""
+
+    speeds: list[int]
+
+    def __post_init__(self) -> None:
+        """Validate inputs."""
+        assert len(self.speeds) > 0
+
+
+@dataclass
+class FixedFanSpeedDataTemplate(
+    BaseDiscoverySchemaDataTemplate, FanSpeedDataTemplate, FixedFanSpeedValueMix
+):
+    """
+    Specifies a fixed set of fan speeds.
+
+    Example:
+      ZWaveDiscoverySchema(
+          platform="fan",
+          hint="configured_fan_speed",
+          ...
+          data_template=FixedFanSpeedDataTemplate(
+              speeds=[32,65,99]
+          ),
+      ),
+
+    `speeds` indicates the maximum setting on the underlying fan controller
+    for each actual speed.
+    """
+
+    def get_speed_config(self, resolved_data: dict[str, Any]) -> list[int]:
+        """Get the fan speed configuration for this device."""
+        return self.speeds
```

Comment:
Are we not testing the fixed fan speed data template? I don't see where this is used in the code either.

Reply:
Yeah, I put it in place because it's going to be needed for a bunch of other fan controllers (e.g., the GE/Jasco 3 speed), but I haven't collected info on their stop-points yet.  I'd previously included tests for it in `discovery_data_template_tests.py`, but you requested I drop those :)

I'm willing to remove it from this PR and add it back in a followup PR once I've collected some information from people who have other fan controllers, if that's preferable.  Let me know what you'd prefer.

Reply:
Yeah, let's remove it for now and add it later when we have real cases including dumps to test with. :+1:

Reply:
Alright, I've pulled it out!